File: features/fixtures/ios-swift-cocoapods/iOSTestApp/scenarios/Scenario.m
//
// Created by Jamie Lynch on 23/03/2018.
// Copyright (c) 2018 Bugsnag. All rights reserved.
//
#import <objc/runtime.h>

#import "Scenario.h"

@implementation Scenario

+ (Scenario *)createScenarioNamed:(NSString *)className
                       withConfig:(BugsnagConfiguration *)config {
    Class clz = NSClassFromString(className);

    if (clz == nil) { // swift class
        clz = NSClassFromString([NSString stringWithFormat:@"iOSTestApp.%@", className]);
    }

    NSAssert(clz != nil, @"Failed to find class named '%@'", className);

    BOOL implementsRun = method_getImplementation(class_getInstanceMethod([Scenario class], @selector(run))) !=
    method_getImplementation(class_getInstanceMethod(clz, @selector(run)));

    NSAssert(implementsRun, @"Class '%@' does not implement the run method", className);

    id obj = [clz alloc];

    NSAssert([obj isKindOfClass:[Scenario class]], @"Class '%@' is not a subclass of Scenario", className);

    return [(Scenario *)obj initWithConfig:config];
}

- (instancetype)initWithConfig:(BugsnagConfiguration *)config {
    if (self = [super init]) {
        self.config = config;
    }
    return self;
}

- (void)run {
}

- (void)startBugsnag {
    [Bugsnag startBugsnagWithConfiguration:self.config];
}

@end

File: features/fixtures/ios-swift-cocoapods/iOSTestApp/scenarios/Scenario.swift
//
// Created by Jamie Lynch on 06/03/2018.
// Copyright (c) 2018 Bugsnag. All rights reserved.
//

import Foundation
import Bugsnag

@objc open class Scenario: NSObject {

    let config: BugsnagConfiguration

    @objc required public init(config: BugsnagConfiguration) {
        self.config = config
    }

    /**
     * Executes the test case
     */
    func run() {}

    func initBugsnag() {
        Bugsnag.start(with: self.config)
    }

}
